File: server/tests/fixtures/alpha_rule_triggers/python/analysis_change_impact.py
# analysis.change_impact fixture
# This file defines a critical class used by many other files
# Should trigger: analysis.change_impact

class CriticalService:
    """A critical service that many other modules depend on."""
    
    def process(self, data):
        """Critical method - changing this impacts many dependents."""
        return self._internal_process(data)
    
    def _internal_process(self, data):
        return {"processed": data}

def critical_function(param):
    """A function that is imported by many modules."""
    service = CriticalService()
    return service.process(param)

File: server/tests/fixtures/alpha_rule_triggers/typescript/analysis_change_impact.ts
// analysis.change_impact fixture
// This file defines a critical class used by many other files
// Should trigger: analysis.change_impact

export class CriticalService {
    process(data: any): object {
        return this.internalProcess(data);
    }
    
    private internalProcess(data: any): object {
        return { processed: data };
    }
}

export function criticalFunction(param: any): object {
    const service = new CriticalService();
    return service.process(param);
}
